src/loading.rs

```diff
@@ -1,14 +1,12 @@
 use crate::entity::Image;
-use futures::Future;
 use gloo_timers::future::TimeoutFuture;
 use seed::prelude::*;
 
 const SLOW_LOADING_THRESHOLD_MS: u32 = 500;
 
-pub fn notify_on_slow_load<Ms>(msg: Ms, error_msg: Ms) -> impl Future<Item = Ms, Error = Ms> {
-    TimeoutFuture::new(SLOW_LOADING_THRESHOLD_MS)
-        .map(|_| msg)
-        .map_err(|_| error_msg)
+pub async fn notify_on_slow_load<Ms>(msg: Ms) -> Result<Ms, Ms> {
```

Comment:
Do you think we will be able to change it in the future to:
```rust
pub async fn notify_on_slow_load<Ms>(msg: Ms) -> Result<Ms, !> {
```
? (https://doc.rust-lang.org/std/primitive.never.html#infallible-errors)

If so, pls add a comment with TODO above this line.

Reply:
This does not work on nightly. Error is
```
   = note: expected enum `std::result::Result<_, !>`
              found enum `std::result::Result<_, page::settings::Msg>`
```